### PR TITLE
run both rbd and cephfs pvc based subscription

### PR DESCRIPTION
**Pull request by par97**

### CODE REVIEW

**Review by ShyamsundarR — COMMENTED** [review]
The workload should decide on the kustomization based on the environment passed to it, rather than passing a kustomiation to the workload.

For example in the case of Deployment we can kustomize 2 items, the StorageClass name and the PVC access mode. These should come from the config and be passed to the Workload class, the Deployment Workload instance will then generate the required kustomization based on these variables and return it when asked for.

This way, we do not need to write each Kustomization as an input to the Workload, but just pass in the config as each Workload knows how to kustomize itself and is unique.

Elaborating this further:
- Add to config an array of storage classes with modes that we want to test from these classes
  - e.g:
    storageClasses:
      - name: rook-ceph-rbd
        mode: RWO
      - name: rook-cephfs
        mode: RWX

- When initializing an instance of the Workload for exhaustive tests, pass additional parameters `StorageClass` and `PVCMode`.

- Now, when the deployer requests a Kustomize from the Workload, return the appropriate kustomization replacing the SC name and PVC mode

- In exhaustive, add another for loop to loop through the storageClasses that we want to test from config

**Comment by ShyamsundarR** [comment]
@par97 one change to this PR as suggested below (only compile tested).

The intention is that we do not need to teach/code the test for each workload, makes it easier to add more tests or test files, instead the deployment is passed the "env" it needs to run on and would know how to best kustomize itself to the env.

Otherwise the PR looks good.

```
diff --git a/e2e/exhaustive_suite_test.go b/e2e/exhaustive_suite_test.go
index 8ef5e52d..34d4cd06 100644
--- a/e2e/exhaustive_suite_test.go
+++ b/e2e/exhaustive_suite_test.go
@@ -40,16 +40,7 @@ func generateWorkloads([]workloads.Workload) {
                        Revision: GITREVISION,
                        AppName:  APPNAME,
                        Name:     fmt.Sprintf("Deployment-%d", i),
-                       Patch: `{
-                                               "patches": [{
-                                                       "target": {
-                                                               "kind": "PersistentVolumeClaim",
-                                                               "name": "busybox-pvc"
-                                                       },
-                                                       "patch": "- op: replace\n  path: /spec/storageClassName\n  value: ` + pvcs[i].StorageClassName +
-                               `\n- op: add\n  path: /spec/accessModes\n  value: [` + pvcs[i].AccessModes + `]"
-                                               }]
-                                       }`,
+                       PVCSpec:  pvcs[i],
                }
                Workloads = append(Workloads, deployment)
        }
diff --git a/e2e/workloads/deployment.go b/e2e/workloads/deployment.go
index 64278d2d..46339afc 100644
--- a/e2e/workloads/deployment.go
+++ b/e2e/workloads/deployment.go
@@ -3,13 +3,15 @@
 
 package workloads
 
+import "github.com/ramendr/ramen/e2e/util"
+
 type Deployment struct {
        // RepoURL  string
        Path     string
        Revision string
        AppName  string
        Name     string
-       Patch    string
+       PVCSpec  util.PVC
 }
 
 func (w Deployment) GetAppName() string {
@@ -33,7 +35,32 @@ func (w Deployment) GetRevision() string {
 }
 
 func (w Deployment) Kustomize() string {
-       return w.Patch
+       if w.PVCSpec.StorageClassName == "" && w.PVCSpec.AccessModes == "" {
+               return ""
+       }
+
+       scName := "rook-ceph-block"
+       if w.PVCSpec.StorageClassName != "" {
+               scName = w.PVCSpec.StorageClassName
+       }
+
+       accessMode := "ReadWriteOnce"
+       if w.PVCSpec.AccessModes != "" {
+               accessMode = w.PVCSpec.AccessModes
+       }
+
+       patch := `{
+               "patches": [{
+                       "target": {
+                               "kind": "PersistentVolumeClaim",
+                               "name": "busybox-pvc"
+                       },
+                       "patch": "- op: replace\n  path: /spec/storageClassName\n  value: ` + scName +
+               `\n- op: add\n  path: /spec/accessModes\n  value: [` + accessMode + `]"
+               }]
+       }`
+
+       return patch
 }
 
 func (w Deployment) GetResources() error {
```

**Review by ShyamsundarR — COMMENTED** [review]
LGTM! Requires a rebase as dependent PRs are merged, and if rebased we can drop the gitignore addition as stated.

File: .gitignore
Comment (par97) [reply .gitignore]: removed this not needed commit.

File: test/addons/argocd/test
Comment (par97) [reply test/addons/argocd/test]: removed this not needed commit and rebased.

**Review by ShyamsundarR — APPROVED** [review]

**Comment by ShyamsundarR** [comment]
Ran this locally on a drenv setup and e2e worked fine. Unsure why e2e failed above, retrying once and potentially merging the PR for subsequent runs.

**Pull request merged**
